docs/content/concepts/sui-move-concepts/collections.mdx

```diff
@@ -1,11 +1,74 @@
-The types and function discussed in this section are built into the Sui framework in modules table and bag. As with dynamic fields, there is also an object_ variant of both: ObjectTable in object_table and ObjectBag in object_bag. The relationship between Table and ObjectTable, and Bag and ObjectBag are the same as between a field and an object field: The former can hold any store type as a value, but objects stored as values are hidden when viewed from external storage. The latter can only store objects as values, but keeps those objects visible at their ID in external storage.
-Current Limitations 
-Collections are built on top of dynamic fields, and so are subject to its limitations. Additionally, the following functionality is planned, but not currently supported:
-sui::bag::contains<K: copy + drop + store>(bag: &Bag, k: K): bool which checks whether a key-value pair exists in bag with key k: K and a value of any type (in addition to contain_with_type which performs a similar check, but requires passing a specific value type).
+Collections provide a convenient way to work with groups of data. The Sui framework provides several modules that provide collection logic:
+
+## bag
+
+A bag is a heterogeneous map-like collection. The collection is similar to `sui::table` in that its keys and values are not stored within the Bag value, but instead are stored using Sui's object system. The Bag struct acts only as a handle into the object system to retrieve those keys and values. This means that Bag values with exactly the same key-value mapping will not be equal, with ==, at runtime.
+
+To learn more, see [Table and Bag](../dynamic-fields/tables-bags.mdx) or the [framework definition](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/docs/bag.md) on GitHub.
+
+## dynamic_field
+
+In addition to the fields declared in its type definition, a Sui object can have dynamic fields that can be added after the object has been constructed. Unlike ordinary field names (which are always statically declared identifiers) a dynamic field name can be any value with the copy, drop, and store abilities (for example, an integer, a boolean, or a string). This gives Sui programmers the flexibility to extend objects on-the-fly, and it also serves as a building block for core collection types.
+
+To learn more, see [Dynamic Fields](../dynamic-fields.mdx) or the [framework definition](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/docs/dynamic_field.md) on GitHub.
+
+## dynamic_object_field
+
+Similar to `sui::dynamic_field`, this module allows for the access of dynamic fields. But unlike, `sui::dynamic_field` the values bound to these dynamic fields must be objects themselves. This allows for the objects to still exist within in storage, which may be important for external tools. The difference is otherwise not observable from within Move.
```

Comment:
```suggestion
Similar to `sui::dynamic_field`, this module allows for the access of dynamic fields. But unlike, `sui::dynamic_field` the values bound to these dynamic fields must be objects themselves. This allows for the objects to still exist within storage, which may be important for external tools. The difference is otherwise not observable from within Move.
```